source/postprocess/boundary_velocity_residual_statistics.cc

```diff
@@ -92,6 +116,26 @@ namespace aspect
 
 
 
+    template <int dim>
+    std::vector<Vector<double> >
```

Comment:
This happens to be the return type in the other function where you want to use the value this function returns, but in reality it is an awkward type for a strain rate that is only necessary because of the constraints of the other postprocessor. But you convert it anyway before returning it (subtracting it from the real strain rate). So I would suggest to return a `Tensor<2,dim-1>` instead. This will also make this function faster, because the size of a Tensor<2,dim-1> is known at compile-time and can be allocated on the stack instead of the heap memory (http://www.programmerinterview.com/data-structures/difference-between-stack-and-heap/).

Reply:
Got it, thanks for explaining!